apps/service_providers/views.py

```diff
@@ -3,35 +3,45 @@
 from django_tables2 import SingleTableView
 
 from ..generics.views import BaseTypeSelectFormView
-from .models import LlmProvider
-from .tables import LlmProviderTable
-from .utils import get_llm_config_form
+from .utils import ServiceProvider, get_service_provider_config_form
 
 
-class LlmProviderTableView(SingleTableView):
+class ServiceProviderMixin:
+    @property
+    def provider_type(self) -> ServiceProvider:
+        type_ = self.kwargs["provider_type"]
```

Comment:
Where does `self.kwargs` come from?

Reply:
They are the standard Django view kwargs.